robot/impl/resource_manager.go

```diff
@@ -211,7 +213,22 @@ func (manager *resourceManager) updateRemoteResourceNames(
 		if _, alreadyCurrent := activeResourceNames[resName]; alreadyCurrent {
 			activeResourceNames[resName] = true
 			if ok && !gNode.IsUninitialized() {
-				continue
+				if !recreateAllClients {
+					continue
+				}
+				if err := manager.markChildrenForUpdate(resName); err != nil {
+					manager.logger.CErrorw(ctx,
+						"failed to mark children of remote for update",
```

Comment:
```suggestion
						"failed to mark children of remote resource for update",
```
I realize you copied from below but it's technically "children of remote resource".